librz/include/rz_arch.h

```diff
@@ -12,7 +12,7 @@
 extern "C" {
 #endif
 
-typedef struct rz_core_t RzCore;
+struct rz_core_t;
```

Comment:
I probably missed the initial arch review, but why does arch needs rzcore? I think there shouldn't be any circular dependencies and arch should be usable even without core headers at all.

Reply:
@XVilka @wargio @officialcjunior

Reply:
It is just for this little API here:
https://www.github.com/rizinorg/rizin/tree/dev/librz%2Finclude%2Frz_arch.h#L43 -- `rz_arch_profile_add_flag_every_io`

which adds the IO/Extended IO registers from the CPU Profiles as flags.

I just thought it doesn't make sense to be put elsewhere. Sorry. 
I think it can be moved to where it should be.

Reply:
We probably could move that initialization in the RzCore instead.

Reply:
Alright.
I will fix that one up in https://github.com/rizinorg/rizin/pull/1211, then.

Reply:
no is better to do this in another PR